src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -475,6 +551,48 @@ int NonClientIslandWindow::_GetResizeHandleHeight() const noexcept
     return HTCAPTION;
 }
 
+// Method Description:
+// - Sets the cursor to the sizing cursor when we hit-test the top sizing border.
+//   We need to do this because we've covered it up with a child window.
+[[nodiscard]] LRESULT NonClientIslandWindow::_OnSetCursor(WPARAM wParam, LPARAM lParam) const noexcept
+{
+    if (LOWORD(lParam) == HTCLIENT)
+    {
+        // Get the cursor position from the _last message_ and not from
+        // `GetCursorPos` (which returns the cursor position _at the
+        // moment_) because if we're lagging behind the cursor's position,
+        // we still want to get the cursor position that was associated
+        // with that message at the time it was sent to handle the message
+        // correctly.
+        const auto screenPtDword = GetMessagePos();
+        POINT screenPt = { GET_X_LPARAM(screenPtDword), GET_Y_LPARAM(screenPtDword) };
+
+        LRESULT hitTest = SendMessage(GetWindowHandle(), WM_NCHITTEST, 0, MAKELPARAM(screenPt.x, screenPt.y));

```

Comment:
Why isn't the pos just forwarded straight into the `lParam` of the `SendMessage` call? It doesn't look like it's used for anything else and this is just pulling it apart to put it back together again, as far as I can read.

Reply:
Great question; I stole this from Greg. It made me want to write a til<->LPARAM conversion but I didn’t realize I’d just be wasting cycles going back and forth.

Reply:
Probably a vestige from when he was doing manual management of the hit testing

Reply:
I don't see any reason why I wrote it like that either. You're probably right, maybe I wanted to see the value in my debugger or I wrote the code differently and I refactored something.